lxd/project/permissions.go

```diff
@@ -1514,3 +1514,115 @@ func AllowSnapshotCreation(p *api.Project) error {
 
 	return nil
 }
+
+// GetRestrictedClusterGroups returns a slice of restricted cluster groups for the given project.
+func GetRestrictedClusterGroups(p *api.Project) []string {
+	return shared.SplitNTrimSpace(p.Config["restricted.cluster.groups"], ",", -1, true)
+}
+
+// AllowClusterMember returns nil if the given project is allowed to use the cluster member.
+func AllowClusterMember(p *api.Project, member *db.NodeInfo) error {
+	clusterGroupsAllowed := GetRestrictedClusterGroups(p)
+
+	if shared.IsTrue(p.Config["restricted"]) && len(clusterGroupsAllowed) > 0 {
+		found := false
+		for _, memberGroupName := range member.Groups {
+			if shared.StringInSlice(memberGroupName, clusterGroupsAllowed) {
+				found = true
+				break
+			}
+		}
+
+		if !found {
+			return fmt.Errorf("Project isn't allowed to use this cluster member")
+		}
+	}
+
+	return nil
+}
+
+// AllowClusterGroup returns nil if the given project is allowed to use the cluster group.
+func AllowClusterGroup(p *api.Project, group string) error {
```

Comment:
Lets call `group` `groupName` to be explicit.

Reply:
Done, also for the function below to be consistent.